### PR TITLE
Use rectangular arrays in System.Console

### PR DESCRIPTION
**Pull request by James-Ko**
Readability improvements to Console.

### CODE REVIEW

File: src/System.Console/src/System/ConsolePal.Unix.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (Maxwe11) [inline src/System.Console/src/System/ConsolePal.Unix.cs]: revert this change

File: src/System.Console/src/System/ConsolePal.Windows.cs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
+// Copyright (c) Microsoft. All rights reserved.
```
Comment (Maxwe11) [inline src/System.Console/src/System/ConsolePal.Windows.cs]: Ditto.

**Comment by stephentoub** [comment]
At the time this code was written (https://github.com/dotnet/corefx/pull/763), multidimensional arrays could not be instantiated on Unix, hence why the code was written to use a jagged array.  That was subsequently fixed (https://github.com/dotnet/coreclr/pull/376).  As such, I'm fine with it being changed to use a multidimensional array now.  But please do revert the unrelated whitespace changes, and please update the commit description; it's not really about readability... I'd prefer to see a description like "Replace jagged with multidim array in Console on Unix", or something like that.

**Comment by James-Ko** [comment]
Looks like my editor stripped the BOM from each of the files- this may be why the first line keeps showing up as part of the commit when there's no visible change.

**Comment by jasonwilliams200OK** [comment]
Regarding BOM, I think we should have a rule at the end of [`.editorconfig`](https://github.com/dotnet/corefx/blob/646cbde026d7b9c2e59374c33f404ac6f38a4643/.editorconfig) file to avoid it in future:

```powershell
# UTF-8 with BOM  
[*.{c,cs,cpp,h,def,txt,xml,json}]  
charset=utf-8-bom 
```

**Comment by sharwell** [comment]
For C# files, you could use the new [SA1412 Store Files as UTF-8](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1412.md) rule in StyleCopAnalyzers (new for Beta 5 which I'm in the process of releasing). :smile:

**Comment by Maxwe11** [comment]
@James-Ko squash into single commit and what about Stephen's comment?

**Comment by James-Ko** [comment]
@sharwell Thanks; I don't often use Visual Studio for minor changes, though, so I created a [shell script](https://gist.github.com/James-Ko/d65e10f6a2aa3fc6ffe0) for people who don't have an IDE.

@Maxwe11 I realize, I was busy for the past few hours; here it is.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
Thanks for updating the PR. LGTM.
